python/pyspark/errors/exceptions/captured.py

```diff
@@ -146,7 +151,7 @@ def getMessage(self) -> str:
         if self._origin is not None and is_instance_of(
             gw, self._origin, "org.apache.spark.SparkThrowable"
         ):
-            errorClass = self._origin.getErrorClass()
+            errorClass = self._origin.getCondition()
```

Comment:
We also need to update an `errorClass` to `condition`, but let me separate it into follow-up tasks as it brings bunch of code changes.

Created umbrella JIRA for error generalization from SPARK-50916, so will put several related tasks there.